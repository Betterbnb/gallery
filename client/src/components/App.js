import React from 'react';
import path from 'path';
import fetch from 'node-fetch';
import {Modal} from './Modal.js';
import styled from 'styled-components';


class App extends React.Component {
	constructor(props) {
		super(props);
		this.state = {
			homeId: '100', //DEFAULTS TO HOME 100
			imagesArr: [{image: null, caption: null}],
			modalDisplay: false
		}

		this.getPics = this.getPics.bind(this);
		this.toggleModal = this.toggleModal.bind(this);
	}

	// getHomeId () {
	// 	let homeId = window.location.href.split('/');
	// 	homeId = homeId[homeId.length-1]; //THIS IS A STRING, NEED TO CONVERT TO INT?
	// 	this.setState({homeId: homeId});
	// }

	getPics() {
		let url = path.join('homes', this.state.homeId, 'images');
		fetch(url)
		.then((res)=>{return res.json()})
		.then((data)=>{
			this.setState({imagesArr: data})
		})
		.catch( (err)=> console.log(err) )
	}

	toggleModal() {
		this.setState({modalDisplay: !this.state.modalDisplay});
	}

	componentDidMount() {
		// this.getHomeId();
		this.getPics();
	}

	render() {
		return (
			<div>
			<h1> Gallery Component </h1>
			<div className="modal">
			<img src={this.state.imagesArr[0].image} onClick={this.toggleModal}/>
			<Modal imagesArr={this.state.imagesArr} modalDisplayState={this.state.modalDisplay} imagesArr={this.state.imagesArr}/>
			</div>
			</div>
		)
	}
}

export default App

//PASS CAPTION INFO INTO PROP