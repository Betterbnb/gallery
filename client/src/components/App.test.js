import React from 'react';
import { shallow, mount, render } from 'enzyme';
import App from './App.js';


describe('App', () => {
  it('should render correctly in "debug" mode', () => {
    const component = shallow(<App debug />);
  
    expect(component).toMatchSnapshot();
  });
});

