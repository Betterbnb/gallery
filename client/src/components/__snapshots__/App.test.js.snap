// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render correctly in "debug" mode 1`] = `
<div>
  <h1>
     Gallery Component 
  </h1>
</div>
`;
